pages/agent/v3/elastic_ci_aws.md.erb

```diff
@@ -0,0 +1,260 @@
+
+If you want to set secrets that your build can access, create a file that sets environment variables and upload it:
+
+```bash
+echo "export MY_ENV_VAR=something secret" > myenv
+aws s3 cp --acl private --sse aws:kms myenv "s3://${SecretsBucket}/env"
+rm myenv
+```
+<div class="Docs__note">
+Currently, you must use the default KMS key for S3. Follow <a href="https://github.com/buildkite/elastic-ci-stack-for-aws/issues/235" target="_blank">issue #235</a> for progress on using specific KMS keys.
+</div>
+
+If you want to store your secrets unencrypted, you can disable it entirely with `BUILDKITE_USE_KMS=false`.
+
+## What's on each machine?
+
+
+* [Amazon Linux 2 LTS](https://aws.amazon.com/amazon-linux-2/)
+* [Buildkite Agent v3.27.0](https://buildkite.com/docs/agent)
+* [Docker](https://www.docker.com) - 19.03.13 (Linux) and 19.03.12 (Windows)
+* [Docker Compose](https://docs.docker.com/compose/) - 1.27.4 (Linux) and 1.27.2 (Windows)
+* [AWS CLI](https://aws.amazon.com/cli/) - useful for performing any ops-related tasks
+<!-- vale off -->
+* [jq](https://stedolan.github.io/jq/) - useful for manipulating JSON responses from CLI tools such as AWS CLI or the Buildkite API
+<!-- vale on -->
+
+## What type of builds does this support?
+
+This stack is designed to run your builds in a share-nothing pattern similar to the [12 factor application principals](http://12factor.net):
+
+* Each project should encapsulate its dependencies through Docker and Docker Compose.
+* Build pipeline steps should assume no state on the machine (and instead rely on [build meta-data](/docs/guides/build-meta-data), [build artifacts](/docs/guides/artifacts) or S3).
+* Secrets are configured via environment variables exposed using the S3 secrets bucket.
+
+By following these conventions you get a scalable, repeatable, and source-controlled CI environment that any team within your organization can use.
+
+## Multiple instances of the stack
+
+If you need different instances sizes and scaling characteristics for different pipelines, you can create multiple stacks. Each can run on a different [Agent queue](/docs/agent/queues), with its own configuration, or even in a different AWS account.
+
+Examples:
+
+* A `docker-builders` stack that provides always-on workers with hot Docker caches (see [Optimizing for slow Docker builds](#optimizing-for-slow-docker-builds))
+* A `pipeline-uploaders` stack with tiny, always-on instances for lightning fast `buildkite-agent pipeline upload` jobs.
+* A `deploy` stack with added credentials and permissions specifically for deployment.
+
+## Autoscaling
+
+If you configure `MinSize` < `MaxSize`, the stack automatically scales up and down based on the number of scheduled jobs.
+
+This means you can scale down to zero when idle, which means you can use larger instances for the same cost.
+
+Metrics are collected with a Lambda function, polling every minute based on the queue the stack is configured with. The autoscaler monitors only one queue.
```

Comment:
> The autoscaler monitors only one queue. 

what are the implications? Can you create a second queue?

Reply:
@yob could you give me more info here?

Reply:
> polling every minute based on the queue the stack is configured with

I think it actually polls every 10s these days, but maybe we can just remove the specificity from the docs?

> The autoscaler monitors only one queue.
>
>> what are the implications? Can you create a second queue?

You can have as many queues as you like on buildkite, but each stack can only monitor a single queue. The montoring is what drives the autoscaling up of the stack, so we only scale up if there's scheduled jobs on one queue.

Maybe that will change one day, but it's not on our short term plans. The implication is that customers will probably want/need one elastic stack per queue.

Reply:
Thanks!